crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1049,6 +1088,7 @@ impl<'db> Type<'db> {
                 // More info in https://docs.python.org/3/library/stdtypes.html#truth-value-testing
                 Truthiness::Ambiguous
             }
+            Type::Type(_) => Truthiness::Ambiguous,
```

Comment:
This is fine for now, but you could possibly add a TODO comment similar to the one in the `Type::ClassLiteral()` branch

Reply:
Ok, I didn't resolve the TODO for now. I'll note it down as a follow-up.

Reply:
Resolving the truthiness of instance types, class-literal types and `SubclassOf` types is _slightly_ more complicated than it seems, because we need to account for the fact that `bool(x)` (where `x` is a member of one of those kinds of types) could fail. We need to emit a diagnostic for things like this:

```pycon
>>> class Bad:
...     __bool__ = None
...     
>>> bool(Bad())
Traceback (most recent call last):
  File "<python-input-1>", line 1, in <module>
    bool(Bad())
    ~~~~^^^^^^^
TypeError: 'NoneType' object is not callable
>>> class BadMetaclass(type):
...     __bool__ = None
...     
>>> class Foo(metaclass=BadMetaclass): pass
... 
>>> bool(Foo)
Traceback (most recent call last):
  File "<python-input-4>", line 1, in <module>
    bool(Foo)
    ~~~~^^^^^
TypeError: 'NoneType' object is not callable
```

That's why we haven't resolved the TODO for the `Type::Instance()` branch here yet either.